src/store/sagas/balances.js

```diff
@@ -296,12 +295,8 @@ function* fetchByAssetSuccess(
 
   // check conditions - network, balance, etc
 
-  const digitalAsset: ExtractReturn<typeof selectDigitalAsset> =
-    yield select(selectDigitalAsset, assetAddress)
-
-  if (!digitalAsset) {
-    throw new Error(t`DigitalAssetNotFoundError`)
-  }
+  const digitalAsset: ExtractReturn<typeof selectDigitalAssetOrThrow> =
+    yield select(selectDigitalAssetOrThrow, assetAddress)
```

Comment:
Do we throw in sagas? How do sagas recover if error is thrown in the middle of state machine transition?

Reply:
good question!
It's ok to `throw` in sagas.
*Service for unhandled exceptions* will care how to handle them after, as I suggested in the confluence doc.